ui/raidboss/emulator/data/PopupTextAnalysis.js

```diff
@@ -1,20 +1,68 @@
 import EmulatorCommon from '../EmulatorCommon';
 import StubbedPopupText from '../overrides/StubbedPopupText';
 
-export default class PopupTextAnalysis extends StubbedPopupText {
-  OnTriggerInternal(trigger, matches, currentTime) {
-    this.currentTriggerStatus = {
-      initialData: EmulatorCommon.cloneData(this.data),
-      condition: undefined,
-      response: false,
-      result: false,
-      delay: false,
-      suppressed: false,
-      executed: false,
-      promise: undefined,
+class Resolver {
+  constructor() {
+    this.promise = null;
+    this.run = null;
+    this.delayTo = null;
+  }
+  async IsResolved(log) {
+    if (this.delayTo) {
+      if (this.delayTo < log.timestamp) {
+        this.delayTo = null;
+        this.delay();
+        await this.delayPromise;
+      } else {
+        return false;
+      }
+    }
+    if (!this.delayTo) {
+      if (this.promise)
+        await this.promise();
+      if (this.run)
+        await this.run();
```

Comment:
Is this asynchronous?

Reply:
I assume you're referencing `this.run` which shouldn't be async, no. Fixing for next commit.